dxr/static/js/dxr.js

```diff
@@ -426,23 +391,30 @@ $(function() {
                     return;
                 }
                 data.query = query;
-                // New results, overwrite
+                // New results, display them.
                 if (myRequestNumber > displayedRequestNumber) {
                     displayedRequestNumber = myRequestNumber;
-                    populateResults(data, false);
-                    var pushHistory = function() {
-                        history.pushState({}, '', queryString);
+                    populateResults(data, appendResults);
+                    var pushHistory = function () {
+                        // Strip off offset= and limit= when updating.
+                        var displayURL = queryString.replace(/&?offset=\d+/, '').replace(/&?limit=\d+/, '');
```

Comment:
The regex should insist on a ? or a & before each param. Otherwise, we'll run into trouble when we inevitably add a foo_offset param later. (Better still might be always building up a query URL from scratch in doQuery, but this is good enough for a quick fix of a user-visible problem.)